### PR TITLE
Vanilla DLC & WS CDLC Vehicles

### PR DESCRIPTION
**Pull request by ante185**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
Adds DLC vehicles and the conditions for them

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
This does not only add Vanilla DLC content but also adds Western Sahara CDLC content.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Apex.sqf [hunk 1]
```diff
@@ -18,8 +18,8 @@
 ["equipmentBox", "Box_NATO_Equip_F"] call _fnc_saveToTemplate; //Changeing this from default will require you to define logistics attachement offset for the box type
 
 ["vehiclesBasic", ["B_T_Quadbike_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightUnarmed", ["B_T_MRAP_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightArmed",["B_T_MRAP_01_hmg_F"]] call _fnc_saveToTemplate;
+["vehiclesLightUnarmed", ["B_T_MRAP_01_F","B_T_MRAP_01_F", "B_T_LSV_01_unarmed_F"]] call _fnc_saveToTemplate;
+["vehiclesLightArmed",["B_T_MRAP_01_hmg_F","B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F", "B_T_MRAP_01_gmg_F", "B_T_LSV_01_AT_F", "B_T_LSV_01_armed_F"]] call _fnc_saveToTemplate;
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Apex.sqf:22]: remove duplicate gmg entry

Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Apex.sqf:22]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_LDF.sqf [hunk 2]
```diff
@@ -76,11 +76,37 @@ private _cargoTrucks = ["I_E_Truck_02_transport_F", "I_E_Truck_02_F"];
...
+if ("ws" in A3A_enabledDLC) then {
     _cargoTrucks = ["I_E_Truck_02_cargo_lxWS","I_E_Truck_02_flatbed_lxWS"];
+	["uavsPortable", ["I_E_UAV_01_F", "I_UAV_02_lxWS"]] call _fnc_saveToTemplate;
 };
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["B_T_AFV_Wheeled_01_cannon_F","B_T_AFV_Wheeled_01_up_cannon_F"]; //Closer to IFV than tank, but no passenger seats.
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["B_T_MRAP_01_F", "B_T_LSV_01_unarmed_F"];
+	_LightArmed append ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F", "B_T_LSV_01_AT_F", "B_T_LSV_01_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_LDF.sqf:88]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_LDF.sqf [hunk 3]
```diff
@@ -18,8 +18,8 @@
 ["equipmentBox", "Box_NATO_Equip_F"] call _fnc_saveToTemplate; //Changeing this from default will require you to define logistics attachement offset for the box type
 
 ["vehiclesBasic", ["I_E_Quadbike_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightUnarmed", ["B_T_MRAP_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightArmed",["B_T_MRAP_01_hmg_F"]] call _fnc_saveToTemplate;
+private _LightUnarmed = ["B_T_MRAP_01_F"];
+private _LightArmed = ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_LDF.sqf:22]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Temperate.sqf [hunk 4]
```diff
@@ -74,11 +74,37 @@ private _cargoTrucks = ["O_T_Truck_02_transport_F", "O_T_Truck_02_F", "O_T_Truck
...
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+	_vehiclesMilitiaCars append ["C_Offroad_01_comms_F", "C_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["O_MBT_04_cannon_F","O_MBT_04_command_F"];
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["O_T_MRAP_02_ghex_F", "O_T_LSV_02_unarmed_F"];
+	_LightArmed append ["O_T_MRAP_02_hmg_ghex_F", "O_T_MRAP_02_gmg_ghex_F", "O_T_LSV_02_AT_F", "O_T_LSV_02_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Temperate.sqf:91]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Enoch.sqf [hunk 5]
```diff
@@ -74,11 +74,37 @@ private _cargoTrucks = ["O_T_Truck_02_transport_F", "O_T_Truck_02_F", "O_T_Truck
...
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+	_vehiclesMilitiaCars append ["C_Offroad_01_comms_F", "C_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["O_MBT_04_cannon_F","O_MBT_04_command_F"];
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["O_T_MRAP_02_ghex_F", "O_T_LSV_02_unarmed_F"];
+	_LightArmed append ["O_T_MRAP_02_hmg_ghex_F", "O_T_MRAP_02_gmg_ghex_F", "O_T_LSV_02_AT_F", "O_T_LSV_02_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Enoch.sqf:91]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Arid.sqf [hunk 6]
```diff
@@ -73,11 +73,36 @@ private _cargoTrucks = ["O_Truck_02_transport_F", "O_Truck_02_covered_F", "O_Tru
...
+
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["O_MBT_04_cannon_F","O_MBT_04_command_F"]; 
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["O_MRAP_02_F", "O_LSV_02_unarmed_F"];
+	_LightArmed append ["O_MRAP_02_hmg_F", "O_MRAP_02_gmg_F", "O_LSV_02_AT_F", "O_LSV_02_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_CSAT_Arid.sqf:89]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Arid.sqf [hunk 7]
```diff
@@ -18,8 +18,8 @@
 ["equipmentBox", "Box_NATO_Equip_F"] call _fnc_saveToTemplate; //Changeing this from default will require you to define logistics attachement offset for the box type
 
 ["vehiclesBasic", ["B_Quadbike_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightUnarmed", ["B_MRAP_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightArmed",["B_MRAP_01_gmg_F", "B_MRAP_01_hmg_F"]] call _fnc_saveToTemplate;
+private _LightUnarmed = ["B_MRAP_01_F"];
+private _LightArmed = ["B_MRAP_01_gmg_F", "B_MRAP_01_hmg_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Arid.sqf:22]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Arid.sqf [hunk 8]
```diff
@@ -75,11 +75,41 @@ private _APCs = ["B_APC_Wheeled_01_cannon_F", "B_APC_Tracked_01_rcws_F"];
...
+};
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["B_AFV_Wheeled_01_cannon_F","B_AFV_Wheeled_01_up_cannon_F"]; //Closer to IFV than tank, but no passenger seats.
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["B_MRAP_01_F","B_LSV_01_unarmed_F"];
+	_LightArmed append ["B_MRAP_01_gmg_F", "B_MRAP_01_hmg_F","B_LSV_01_AT_F","B_LSV_01_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Arid.sqf:92]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf [hunk 9]
```diff
@@ -18,8 +18,8 @@
 ["equipmentBox", "Box_NATO_Equip_F"] call _fnc_saveToTemplate; //Changeing this from default will require you to define logistics attachement offset for the box type
 
 ["vehiclesBasic", ["B_T_Quadbike_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightUnarmed", ["B_T_MRAP_01_F", "B_T_MRAP_01_F", "B_T_LSV_01_unarmed_F"]] call _fnc_saveToTemplate;
-["vehiclesLightArmed", ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F", "B_T_MRAP_01_gmg_F", "B_T_LSV_01_AT_F", "B_T_LSV_01_armed_F"]] call _fnc_saveToTemplate; //this line determines light and armed vehicles -- Example: ["vehiclesLightArmed", ["B_MRAP_01_hmg_F", "B_MRAP_01_gmg_F"]] -- Array, can contain multiple assets
+private _LightUnarmed = ["B_T_MRAP_01_F"];
+private _LightArmed = ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F"]; //this line determines light and armed vehicles -- Example: ["vehiclesLightArmed", ["B_MRAP_01_hmg_F", "B_MRAP_01_gmg_F"]] -- Array, can contain multiple assets
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf:22]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf [hunk 10]
```diff
@@ -75,10 +75,42 @@ private _APCs = ["B_T_APC_Wheeled_01_cannon_F", "B_T_APC_Tracked_01_rcws_F"];
...
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+	_vehiclesMilitiaCars append ["C_Offroad_01_comms_F", "C_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["B_T_AFV_Wheeled_01_cannon_F","B_T_AFV_Wheeled_01_up_cannon_F"]; //Closer to IFV than tank, but no passenger seats.
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["B_T_MRAP_01_F", "B_T_LSV_01_unarmed_F"];
+	_LightArmed append ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F", "B_T_LSV_01_AT_F", "B_T_LSV_01_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Temperate.sqf:92]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Tropical.sqf [hunk 11]
```diff
@@ -18,8 +18,8 @@
 ["equipmentBox", "Box_NATO_Equip_F"] call _fnc_saveToTemplate; //Changeing this from default will require you to define logistics attachement offset for the box type
 
 ["vehiclesBasic", ["B_T_Quadbike_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightUnarmed", ["B_T_MRAP_01_F"]] call _fnc_saveToTemplate;
-["vehiclesLightArmed",["B_T_MRAP_01_hmg_F"]] call _fnc_saveToTemplate;
+private _LightUnarmed = ["B_T_MRAP_01_F"];
+private _LightArmed = ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Tropical.sqf:22]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

File: A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Tropical.sqf [hunk 12]
```diff
@@ -76,10 +76,43 @@ private _APCs = ["B_T_APC_Wheeled_01_cannon_F", "B_T_APC_Tracked_01_rcws_F"];
...
+
+if ("enoch" in A3A_enabledDLC) then {
+	_vehiclesPolice append ["B_GEN_Offroad_01_comms_F","B_GEN_Offroad_01_covered_F"];
+};
+if ("tanks" in A3A_enabledDLC) then {
+	_Tanks append ["B_T_AFV_Wheeled_01_cannon_F","B_T_AFV_Wheeled_01_up_cannon_F"]; //Closer to IFV than tank, but no passenger seats.
+};
+if ("expansion" in A3A_enabledDLC) then {
+	_LightUnarmed append ["B_T_MRAP_01_F", "B_T_LSV_01_unarmed_F"];
+	_LightArmed append ["B_T_MRAP_01_hmg_F", "B_T_MRAP_01_gmg_F", "B_T_LSV_01_AT_F", "B_T_LSV_01_armed_F"];
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/Templates/Vanilla/Vanilla_AI_NATO_Tropical.sqf:92]: Please duplicate the non-gmg entries so gmg has a lower chance spawning as it's OP.

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]

**Comment by ante185** [comment]
in _Vanilla_Reb_Vehicle_Attributes_ I moved the western sahara uparmoured offroads into a if statement based of whether western sahara is loaded or not, as these would cause errors otherwise.

**Comment by ante185** [comment]
GMG likely hood should be sufficiently reduced/addressed now.
1 in 3 for vanilla, 1 in 5 with apex enabled.

**Review by Bob-Murphy — APPROVED** [review]

**Review by Bob-Murphy — APPROVED** [review]

**Pull request merged**